diff -Naur mock-urpm.orig/py/mock_urpm/backend.py mock-urpm/py/mock_urpm/backend.py
--- mock-urpm.orig/py/mock_urpm/backend.py	2015-01-23 08:10:08.253389456 -0500
+++ mock-urpm/py/mock_urpm/backend.py	2015-01-23 07:10:04.000000000 -0500
@@ -198,9 +198,16 @@
     def _unlock_and_rm_chroot(self):
         if not os.path.exists(self.basedir):
             return
+
         t = self.basedir + ".tmp"
         if os.path.exists(t):
+            for cmd in reversed(self.umountCmds):
+                try:
+                    mock_urpm.util.do(cmd, raiseExc=1, shell=True, verbose=self.verbose)
+                except mock_urpm.exception.Error, e:
+                    pass
             mock_urpm.util.rmtree(t, selinux=self.selinux)
+
         os.rename(self.basedir, t)
         self.buildrootLock.close()
         try:
