diff -Nur mock-urpm.old/etc/mock-urpm/Mandriva-2011-i586.cfg mock-urpm/etc/mock-urpm/Mandriva-2011-i586.cfg
--- mock-urpm.old/etc/mock-urpm/Mandriva-2011-i586.cfg	2013-02-12 15:21:34.000000000 +0400
+++ mock-urpm/etc/mock-urpm/Mandriva-2011-i586.cfg	2013-10-01 12:59:31.594365143 +0400
@@ -4,6 +4,8 @@
 
 config_opts['chroot_setup'] = 'basesystem-minimal locales locales-en locales-de locales-uk locales-es locales-ru basesystem-minimal libmpc2 libmpfr4 libnatspec0 libpwl5 make patch unzip mandriva-release-common binutils curl gcc gcc-c++ gnupg mandriva-release-Free rpm-build'
 config_opts['urpmi_options'] = '--no-suggests --no-verify-rpm'
+config_opts['urpm_options'] = ''
+
 
 # If it's True - current urpmi configs will be copied to the chroot.
 # Ater that other media will be added.
diff -Nur mock-urpm.old/etc/mock-urpm/Mandriva-2011-x86_64.cfg mock-urpm/etc/mock-urpm/Mandriva-2011-x86_64.cfg
--- mock-urpm.old/etc/mock-urpm/Mandriva-2011-x86_64.cfg	2013-02-12 15:21:34.000000000 +0400
+++ mock-urpm/etc/mock-urpm/Mandriva-2011-x86_64.cfg	2013-10-01 12:59:34.859365241 +0400
@@ -4,6 +4,7 @@
 
 config_opts['chroot_setup'] = 'basesystem-minimal locales locales-en locales-de locales-uk locales-es locales-ru basesystem-minimal lib64mpc2 lib64mpfr4 lib64natspec0 lib64pwl5 make patch unzip mandriva-release-common binutils curl gcc gcc-c++ gnupg mandriva-release-Free rpm-build'
 config_opts['urpmi_options'] = '--no-suggests --no-verify-rpm'
+config_opts['urpm_options'] = ''
 
 
 # If it's True - current urpmi configs will be copied to the chroot.
diff -Nur mock-urpm.old/etc/mock-urpm/Mandriva-cooker-i586.cfg mock-urpm/etc/mock-urpm/Mandriva-cooker-i586.cfg
--- mock-urpm.old/etc/mock-urpm/Mandriva-cooker-i586.cfg	2013-02-12 15:21:34.000000000 +0400
+++ mock-urpm/etc/mock-urpm/Mandriva-cooker-i586.cfg	2013-10-01 12:59:38.450365348 +0400
@@ -4,6 +4,7 @@
 
 config_opts['chroot_setup'] = 'basesystem-minimal locales locales-en locales-de locales-uk locales-es locales-ru basesystem-minimal libmpc3 libmpfr4 libnatspec0 libpwl5 make patch unzip mandriva-release-common binutils curl gcc gcc-c++ gnupg rpm-build'
 config_opts['urpmi_options'] = '--no-suggests --no-verify-rpm'
+config_opts['urpm_options'] = ''
 
 # If it's True - current urpmi configs will be copied to the chroot.
 # Ater that other media will be added.
diff -Nur mock-urpm.old/etc/mock-urpm/Mandriva-cooker-x86_64.cfg mock-urpm/etc/mock-urpm/Mandriva-cooker-x86_64.cfg
--- mock-urpm.old/etc/mock-urpm/Mandriva-cooker-x86_64.cfg	2013-02-12 15:21:34.000000000 +0400
+++ mock-urpm/etc/mock-urpm/Mandriva-cooker-x86_64.cfg	2013-10-01 12:59:41.665365444 +0400
@@ -5,6 +5,7 @@
 config_opts['chroot_setup'] = 'basesystem-minimal locales locales-en locales-de locales-uk locales-es locales-ru basesystem-minimal lib64mpc3 lib64mpfr4 lib64natspec0 lib64pwl5 make patch unzip mandriva-release-common binutils curl gcc gcc-c++ gnupg rpm-build'
 
 config_opts['urpmi_options'] = '--no-suggests --no-verify-rpm'
+config_opts['urpm_options'] = ''
 
 
 # If it's True - current urpmi configs will be copied to the chroot.
diff -Nur mock-urpm.old/etc/mock-urpm/Rosa-2012.1-i586.cfg mock-urpm/etc/mock-urpm/Rosa-2012.1-i586.cfg
--- mock-urpm.old/etc/mock-urpm/Rosa-2012.1-i586.cfg	2013-02-12 15:21:34.000000000 +0400
+++ mock-urpm/etc/mock-urpm/Rosa-2012.1-i586.cfg	2013-10-01 12:59:44.906365540 +0400
@@ -4,6 +4,7 @@
 
 config_opts['chroot_setup'] = 'basesystem-minimal locales locales-en locales-de locales-uk locales-es locales-ru basesystem-minimal libmpc2 libmpfr4 libnatspec0 libpwl5 make patch unzip mandriva-release-common binutils curl gcc gcc-c++ gnupg rpm-build'
 config_opts['urpmi_options'] = '--no-suggests --no-verify-rpm'
+config_opts['urpm_options'] = ''
 
 # If it's True - current urpmi configs will be copied to the chroot.
 # Ater that other media will be added.
diff -Nur mock-urpm.old/etc/mock-urpm/Rosa-2012.1-x86_64.cfg mock-urpm/etc/mock-urpm/Rosa-2012.1-x86_64.cfg
--- mock-urpm.old/etc/mock-urpm/Rosa-2012.1-x86_64.cfg	2013-02-12 15:21:34.000000000 +0400
+++ mock-urpm/etc/mock-urpm/Rosa-2012.1-x86_64.cfg	2013-10-01 12:59:48.314365642 +0400
@@ -4,6 +4,7 @@
 
 config_opts['chroot_setup'] = 'basesystem-minimal locales locales-en locales-de locales-uk locales-es locales-ru basesystem-minimal lib64mpc2 lib64mpfr4 lib64natspec0 lib64pwl5 make patch unzip mandriva-release-common binutils curl gcc gcc-c++ gnupg rpm-build'
 config_opts['urpmi_options'] = '--no-suggests --no-verify-rpm'
+config_opts['urpm_options'] = ''
 
 # If it's True - current urpmi configs will be copied to the chroot.
 # Ater that other media will be added.
diff -Nur mock-urpm.old/etc/mock-urpm/Rosa-2012lts-i586.cfg mock-urpm/etc/mock-urpm/Rosa-2012lts-i586.cfg
--- mock-urpm.old/etc/mock-urpm/Rosa-2012lts-i586.cfg	2013-02-12 15:21:34.000000000 +0400
+++ mock-urpm/etc/mock-urpm/Rosa-2012lts-i586.cfg	2013-10-01 12:59:51.465365736 +0400
@@ -4,6 +4,7 @@
 
 config_opts['chroot_setup'] = 'basesystem-minimal locales locales-en locales-de locales-uk locales-es locales-ru basesystem-minimal libmpc2 libmpfr4 libnatspec0 libpwl5 make patch unzip mandriva-release-common binutils curl gcc gcc-c++ gnupg mandriva-release-Free rpm-build'
 config_opts['urpmi_options'] = '--no-suggests --no-verify-rpm'
+config_opts['urpm_options'] = ''
 
 # If it's True - current urpmi configs will be copied to the chroot.
 # Ater that other media will be added.
diff -Nur mock-urpm.old/etc/mock-urpm/Rosa-2012lts-x86_64.cfg mock-urpm/etc/mock-urpm/Rosa-2012lts-x86_64.cfg
--- mock-urpm.old/etc/mock-urpm/Rosa-2012lts-x86_64.cfg	2013-02-12 15:21:34.000000000 +0400
+++ mock-urpm/etc/mock-urpm/Rosa-2012lts-x86_64.cfg	2013-10-01 12:59:53.682365802 +0400
@@ -4,6 +4,7 @@
 
 config_opts['chroot_setup'] = 'basesystem-minimal locales locales-en locales-de locales-uk locales-es locales-ru basesystem-minimal lib64mpc2 lib64mpfr4 lib64natspec0 lib64pwl5 make patch unzip mandriva-release-common binutils curl gcc gcc-c++ gnupg mandriva-release-Free rpm-build'
 config_opts['urpmi_options'] = '--no-suggests --no-verify-rpm'
+config_opts['urpm_options'] = ''
 
 # If it's True - current urpmi configs will be copied to the chroot.
 # Ater that other media will be added.
diff -Nur mock-urpm.old/py/mock_urpm/backend.py mock-urpm/py/mock_urpm/backend.py
--- mock-urpm.old/py/mock_urpm/backend.py	2013-02-12 15:21:34.000000000 +0400
+++ mock-urpm/py/mock_urpm/backend.py	2013-10-01 13:11:41.576386897 +0400
@@ -84,6 +84,7 @@
         self.use_system_media = config['use_system_media']
         self.urpmi_config_dir = config['urpmi_config_dir']
         self.urpmi_options = config['urpmi_options']
+        self.urpm_options = config['urpm_options']
         #self.builddep_path = config['urpmi_path']
         self.macros = config['macros']
         self.more_buildreqs = config['more_buildreqs']
@@ -318,6 +319,7 @@
             
             self.root_log.debug("Adding media...")
             urpmicmd = [self.urpmi_addmedia_path]
+            urpmicmd.extend(self.urpm_options.split())
             urpmicmd.extend(('--urpmi-root', self.makeChrootPath()))
             
             for medium in self.urpmi_media:
@@ -336,6 +338,7 @@
                     raise mock_urpm.exception.UrpmiError, str(e)
             
             c = ['urpmi.update', '-a',  '--urpmi-root', self.makeChrootPath()]
+            c.extend(self.urpm_options.split())
             mock_urpm.util.do(c, returnOutput=1, verbose=self.verbose)
             
         
